sdk/src/main/java/io/customer/sdk/CustomerIO.kt

```diff
@@ -350,6 +350,10 @@ class CustomerIO internal constructor(
                 // we need to unregister the previous activity lifecycle because it doesn't get garbage collected
                 // and will continue to hold a reference to the previous app context and return callbacks
                 appContext.unregisterActivityLifecycleCallbacks(it.diGraph.activityLifecycleCallbacks)
+                // Cancelling queue timer as the new queue timer will take care of running queue tasks.
+                // If we do not cancel old timer, it results in multiple timers being run and accessing
+                // the same tasks.
+                it.diGraph.queue.cancelTimer()
```

Comment:
Thanks for adding this along with a comment explaining why this is here. 

In a call together, we talked about a long-term solution where `CustomerIO.Builder.build()` can be called 2+ times and the SDK config is the only thing that gets modified but the SDK instance or any dependencies do not get modified. 

Do we have a ticket to capture that refactor? I want to make sure before we introduce code like this that we need to maintain in the meantime before we do the refactor.

Reply:
Created a ticket for this work https://github.com/customerio/issues/issues/9915